apps/src/p5lab/gamelab/GameLab.js

```diff
@@ -23,4 +24,35 @@ GameLab.prototype.init = function(config) {
   return P5Lab.prototype.init.call(this, config);
 };
 
+GameLab.prototype.resetHandler = function(ignore) {
+  $('.droplet-main-canvas').css('background-color', '#FFF');
+  $('.droplet-transition-container').css('background-color', '#FFF');
+  $('.ace_scroller').css('background-color', '#FFF');
+  this.reset();
+};
+
+GameLab.prototype.runButtonClick = function() {
+  $('.droplet-main-canvas').css('background-color', '#E5E5E5');
+  $('.droplet-transition-container').css('background-color', '#E5E5E5');
+  $('.ace_scroller').css('background-color', '#E5E5E5');
+  this.studioApp_.toggleRunReset('reset');
+  // document.getElementById('spinner').style.visibility = 'visible';
+  if (this.studioApp_.isUsingBlockly()) {
+    Blockly.mainBlockSpace.traceOn(true);
+  }
+  this.studioApp_.attempts++;
+  this.execute();
+
+  // Enable the Finish button if is present:
+  var shareCell = document.getElementById('share-cell');
+  if (shareCell && !this.level.validationCode) {
+    shareCell.className = 'share-cell-enabled';
+
+    // Adding completion button changes layout.  Force a resize.
+    this.studioApp_.onResize();
+  }
+
+  postContainedLevelAttempt(this.studioApp_);
+};
```

Comment:
To avoid duplicating the code that gamelab and spritelab both use, can you do something like this?

```suggestion
P5Lab.prototype.runButtonClick.call(this);
```